src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const result = [];
+
+  for (let i = 0; i < actions.length; i++) {
```

Comment:
Indexes are not needed here, so it will be better to use the for of loop.
Also use destruction for `action` after
```suggestion
  for (const action of actions) {
    const { type, ........... } = action;
```